lib/pbench/test/unit/server/test_compare_datasets.py

```diff
@@ -0,0 +1,150 @@
+
+        response = query_get_as(["uperf_1", "uperf_2"], "drb", HTTPStatus.FORBIDDEN)
+        assert response.json == {
+            "message": "User drb is not authorized to READ a resource owned by test with private access"
+        }
+
+    def test_unsupported_benchmark(self, query_get_as, monkeypatch):
+        extract_data_called = False
+
+        def mock_compare_csv_to_json(*args, **kwargs):
+            nonlocal extract_data_called
+            extract_data_called = True
+
+        def mock_get_value(dataset, key) -> str:
+            return "hammerDB"
+
+        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
+        monkeypatch.setattr(
+            QuisbyProcessing, "compare_csv_to_json", mock_compare_csv_to_json
+        )
+        response = query_get_as(
+            ["uperf_1", "uperf_2"], "test", HTTPStatus.UNSUPPORTED_MEDIA_TYPE
+        )
+        assert response.json["message"] == "Unsupported Benchmark: hammerDB"
+        assert not extract_data_called
+
+    def test_successful_get(self, query_get_as, monkeypatch):
+        def mock_compare_csv_to_json(
+            self, benchmark_name, input_type, data_stream
+        ) -> JSON:
+            return {"status": "success", "json_data": "quisby_data"}
+
+        monkeypatch.setattr(CacheManager, "find_dataset", self.mock_find_dataset)
+        monkeypatch.setattr(Metadata, "getvalue", self.mock_get_value)
+        monkeypatch.setattr(
+            QuisbyProcessing, "compare_csv_to_json", mock_compare_csv_to_json
+        )
+
+        response = query_get_as(["uperf_1", "uperf_2"], "test", HTTPStatus.OK)
+        assert response.json["status"] == "success"
+        assert response.json["json_data"] == "quisby_data"
+
+    def test_unsuccessful_get_with_incorrect_data(self, query_get_as, monkeypatch):
+        def mock_find_dataset_with_incorrect_data(self, dataset) -> Tarball:
+            class Tarball(object):
+                tarball_path = Path("/dataset/tarball.tar.xz")
+
+                def extract(tarball_path, path) -> str:
+                    return "IncorrectData"
+
+            return Tarball
+
+        def mock_compare_csv_to_json(
+            self, benchmark_name, input_type, data_stream
+        ) -> JSON:
+            return {"status": "failed", "exception": "Unsupported Media Type"}
+
+        monkeypatch.setattr(
+            CacheManager, "find_dataset", mock_find_dataset_with_incorrect_data
+        )
+        monkeypatch.setattr(Metadata, "getvalue", self.mock_get_value)
+        monkeypatch.setattr(QuisbyProcessing, "extract_data", mock_compare_csv_to_json)
+        response = query_get_as(
+            ["uperf_1", "uperf_2"], "test", HTTPStatus.INTERNAL_SERVER_ERROR
+        )
+        assert response.json["message"].startswith(
+            "Internal Pbench Server Error: log reference "
+        )
```

Comment:
I can't really explain how (the fact that line 133 is returning a class object instead of an instance might be contributing to it, although the botched mock at line 144 is probably the source...) but this test isn't working.  That is, looking at [the coverage report](http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-3470/4/cobertura/server_api_resources/compare_dataset_py/#137), the `raise APIInternalError` at [`compare_dataset.py` line 137](https://github.com/distributed-system-analysis/pbench/blob/ceff0e4e54e6b131d4c75941699fae82c2f4d53f/lib/pbench/server/api/resources/compare_dataset.py#L137) is never executed...so the assertion here at line 148 is passing on a _different_ internal error....